lib/tensorflow/session.rb

```diff
@@ -121,4 +51,122 @@ def extend_graph(graph)
   	Tensorflow::TF_ExtendGraph(self.session, graph_def_to_c_array(graph.graph_def_raw), graph.graph_def_raw.length, self.status)
   	self.graph = graph
   end
+
+  private
+
+  def initialize_inputs(inputs)
+    input_names = Tensorflow::String_Vector.new
+    input_values = Tensorflow::Tensor_Vector.new
+    if inputs != nil
+      inputs.each do |key, value|
+        input_values.push(value)
+        input_names.push(key)
+      end
+    end
+
+    return input_names, input_values
+  end
+
+  def initialize_outputs(outputs)
+    output_names = Tensorflow::String_Vector.new
+    outputs.each do |name|
+      output_names.push(name)
+    end
+
+    output_values = Tensorflow::Tensor_Vector.new
+
+    return output_names, output_values
+  end
+
+  def initialize_targets(targets)
+    target_names = Tensorflow::String_Vector.new
+    if targets != nil
+      targets.each do |name|
+        target_names.push(name)
+      end
+    end
+
+    target_names
+  end
+
+  def convert_value_for_output_array(value)
+    size = Tensorflow::tensor_size(value)
+    c_array = construct_c_array(value, size)
+    length_by_dimension = length_by_dimension(value)
+    arrange_into_dimensions(c_array, size, length_by_dimension)
+  end
+
+  # Returns an array containing the length of the tensor in each dimension
+  def length_by_dimension(value)
+    num_dimensions = Tensorflow::TF_NumDims(value)
+    result = []
+
+    (0..num_dimensions - 1).each_with_object([]) do |dimension, array|
+      array.push(Tensorflow::TF_Dim(value, dimension))
+    end
```

Comment:
Couldn't it just be `(0..num_dimensions - 1).map { |dimension| Tensorflow::TF_Dim(value, dimension) }`?
Or am I missing something?